OutOfSchool/OutOfSchool.BusinessLogic/Services/Database/WorkshopService.cs

```diff
@@ -631,6 +648,121 @@ public async Task<SearchResult<WorkshopCard>> GetNearestByFilter(WorkshopFilter
         return result;
     }
 
+    /// <inheritdoc/>
+    /// <exception cref="InvalidOperationException">If the region admin is not found in the database.</exception>
+    public async Task<SearchResult<WorkshopDto>> FetchByFilterForAdmins(WorkshopFilterAdministration filter = null)
+    {
+        logger.LogInformation("Started retrieving Workshops by filter for admins.");
+        filter = filter ?? new WorkshopFilterAdministration();
+
+        var predicate = await PredicateBuildForAdminds(filter);
+
+        var workshops = await workshopRepository.Get(
+                skip: filter.From,
+                take: filter.Size,
+                includeProperties: includingPropertiesForMappingDtoModel,
+                whereExpression: predicate,
+                asNoTracking: true)
+            .ToListAsync()
+            .ConfigureAwait(false);
+
+        var workshopsDTO = mapper.Map<List<WorkshopDto>>(workshops);
+        var workshopsCount = workshops.Count;
+
+        logger.LogInformation("Retrieved {WorkshopsCount} matching recods by filter for admins.", workshopsCount);
```

Comment:
In case when mapper fail - this log will be lost. I'd suggest to move it to the 668 line

Reply:
fixed